src/components/Common/RotatingText.tsx

```diff
@@ -0,0 +1,271 @@
+"use client";
+
+import {
+  AnimatePresence,
+  motion,
+  Transition,
+  type Target,
+  type TargetAndTransition,
+  type VariantLabels,
+} from "motion/react";
+import React, {
+  forwardRef,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useState,
+} from "react";
+
+import "./RotatingText.css";
+
+function cn(...classes: (string | undefined | null | boolean)[]): string {
+  return classes.filter(Boolean).join(" ");
+}
+
+export interface RotatingTextRef {
+  next: () => void;
+  previous: () => void;
+  jumpTo: (index: number) => void;
+  reset: () => void;
+}
+
+export interface RotatingTextProps extends Omit<
+  React.ComponentPropsWithoutRef<typeof motion.span>,
+  "children" | "transition" | "initial" | "animate" | "exit"
+> {
+  texts: string[];
+  transition?: Transition;
+  initial?: boolean | Target | VariantLabels;
+  animate?: boolean | VariantLabels | TargetAndTransition;
+  exit?: Target | VariantLabels;
+  animatePresenceMode?: "sync" | "wait";
+  animatePresenceInitial?: boolean;
+  rotationInterval?: number;
+  staggerDuration?: number;
+  staggerFrom?: "first" | "last" | "center" | "random" | number;
+  loop?: boolean;
+  auto?: boolean;
+  splitBy?: string;
+  onNext?: (index: number) => void;
+  mainClassName?: string;
+  splitLevelClassName?: string;
+  elementLevelClassName?: string;
+}
+
+const RotatingText = forwardRef<RotatingTextRef, RotatingTextProps>(
```

Comment:
forwardRef is not needed anymore/deprecated in 19 

https://react.dev/blog/2024/12/05/react-19#ref-as-a-prop